libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificateBlacklist.java

```diff
@@ -0,0 +1,213 @@
+
+    if (directory.isDirectory()) {
+      File file = new File(directory, BLACKLIST_FILE);
+
+      if (file.exists()) {
+        try {
+          blacklist = obtainBlacklistContents(file);
+          blacklist.remove(BLACKLIST_HEAD);
+        } catch (IOException exception) {
+          Log.e(LOG_TAG, RETRIEVE_BLACKLIST_FAIL, exception);
+        }
+      }
+    }
+
+    return blacklist;
+  }
+
+  boolean daySinceLastUpdate() {
+    long millisecondDiff = System.currentTimeMillis() - retrieveLastUpdateTime();
+    long dayInMilliseconds = 86400000;
+
+    return millisecondDiff >= dayInMilliseconds;
+  }
+
+  private long retrieveLastUpdateTime() {
+    File directory = context.getFilesDir();
+    File file = new File(directory, BLACKLIST_FILE);
+
+    long lastUpdateTime = 0;
+
+    if (file.exists()) {
+      try {
+        List<String> blacklist = obtainBlacklistContents(file);
+        lastUpdateTime = Long.valueOf(blacklist.get(BLACKLIST_HEAD));
+      } catch (IOException exception) {
+        Log.e(LOG_TAG, RETRIEVE_TIME_FAIL, exception);
+      }
+    }
+
+    return lastUpdateTime;
+  }
+
+  void updateBlacklist() {
+    HttpUrl requestUrl = new HttpUrl.Builder().scheme(HTTPS_SCHEME)
+      .host(determineConfigEndpoint())
+      .addQueryParameter(ACCESS_TOKEN_QUERY_PARAMETER, accessToken)
+      .build();
+
+    Request request = new Request.Builder()
+      .url(requestUrl)
+      .build();
+
+    OkHttpClient client = new OkHttpClient();
+    client.newCall(request).enqueue(this);
+  }
+
+  private void saveBlackList(List<String> revokedKeys) {
+    String fileContents = createListContent(revokedKeys);
+    FileOutputStream outputStream;
+
+    try {
+      outputStream = context.openFileOutput(BLACKLIST_FILE, Context.MODE_PRIVATE);
+      outputStream.write(fileContents.getBytes());
+      outputStream.close();
+    } catch (Exception exception) {
+      Log.e(LOG_TAG, SAVE_BLACKLIST_FAIL, exception);
+    }
+  }
+
+  @Override
+  public void onFailure(Call call, IOException exception) {
+    Log.e(LOG_TAG, REQUEST_FAIL, exception);
+  }
+
+  @Override
+  public void onResponse(Call call, Response response) throws IOException {
+    List<String> revokedKeys = extractResponse(response);
+
+    saveBlackList(revokedKeys);
+  }
+
+  private String createListContent(List<String> revokedKeys) {
+    Date date = new Date();
+
+    StringBuilder content = new StringBuilder(date.getTime() + NEW_LINE);
+
+    for (String key : revokedKeys) {
+      content.append(SHA256).append(key).append(NEW_LINE);
+    }
+
+    return content.toString();
+  }
+
+  private List<String> obtainBlacklistContents(File file) throws IOException {
+    InputStream inputStream = new FileInputStream(file);
+    BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+
+    List<String> blacklist = new ArrayList<>();
+
+    boolean done = false;
+
+    while (!done) {
+      String line = reader.readLine();
```

Comment:
reader.readLine() might throw an IOException, then 
`reader.close();
  inputStream.close();`
block will not be called on line 175 and 176. right? shouldn't we handle IOException internally and call close() in finally block? one option is to return an empty list if any of the I/O operations fail. thoughts?